### PR TITLE
[CPU]Add cache support to transpose node

### PR DESCRIPTION
**Pull request by usstq**
### Details:
 - *Add cache support to transpose node*
 - *...*

### Tickets:
 - **

### CODE REVIEW

**Comment by usstq** [comment]
@dmitry-gorokhov  Hi, can you please review this PR? Thanks!

**Comment by dmitry-gorokhov** [comment]
@maxnick please review the PR

File: src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp [hunk 1]
```diff
@@ -1,323 +1,327 @@
...
+        InputShape inputShapes;
+        std::vector<size_t> inputOrder;
+        std::string targetDevice;
+        CPUSpecificParams cpuParams;
+        std::map<std::string, std::string> additionalConfig;
+        std::tie(inputShapes, inputOrder, netPrecision, targetDevice, additionalConfig, cpuParams) = obj.param;
+
+        std::ostringstream result;
+        result << "IS=" << CommonTestUtils::partialShape2str({inputShapes.first}) << "_";
+        result << "TS=" << CommonTestUtils::vec2str(inputShapes.second) << "_";
```
Comment (maxnick) [inline src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp:39]: The second part of the `inputShapes` pair is a vector of vectors, so we should print it using loop
```cpp
        result << "TS=(";
        for (const auto& shape : inputShapes.second) {
            result << CommonTestUtils::vec2str(shape) << "_";
        }
        result << ")_";
```

File: src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp [hunk 2]
```diff
@@ -1,323 +1,327 @@
...
+
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        selectedType = makeSelectedTypeStr("unknown", inType);
+
+        targetStaticShapes.reserve(inputShapes.second.size());
+        for (const auto& staticShape : inputShapes.second) {
+            targetStaticShapes.push_back({staticShape});
+        }
+        inputDynamicShapes = { inputShapes.first };
```
Comment (maxnick) [inline src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp:67]: It is better to use `init_input_shapes({inputShape});`

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp (hunk 1 above)
Comment (usstq) [reply src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp:39]: OK, I will change.
FYI :
 there is a vec2str overload which takes vector of vectors as input: 
```cpp
template<typename vecElementType>
inline std::string vec2str(const std::vector<std::vector<vecElementType>> &vec) {
    std::ostringstream result;
    for (const auto &v : vec) {
        result << vec2str<vecElementType>(v);
    }
    return result.str();
}
```

The only difference is it didn't use "_" to separate shapes and lack of outermost braces `()`.

File: src/plugins/intel_cpu/src/nodes/mkldnn_transpose_node.cpp [hunk 3]
```diff
@@ -127,7 +161,21 @@ void MKLDNNTransposeNode::prepareParams() {
         params.order.assign(orderPtr, orderPtr + orderLen);
     }
 
-    execPtr = std::make_shared<TransposeJitExecutor>(params);
+    TransposeKey key = {params};
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/mkldnn_transpose_node.cpp:164]: Good idea, changed, Thanks!

File: src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp (hunk 2 above)
Comment (usstq) [reply src/tests/functional/plugin/cpu/single_layer_tests/transpose.cpp:67]: changed, Thanks!

**Review by maxnick — APPROVED** [review]

**Pull request merged**